### PR TITLE
replica: add compatibility check and handling for ReplicaAdd

### PR DESCRIPTION
**Pull request by derekbit**
[Longhorn 4705](https://github.com/longhorn/longhorn/issues/4705)

Signed-off-by: Derek Su <derek.su@suse.com>

### CODE REVIEW

File: controller/engine_controller.go [hunk 1]
```diff
@@ -1541,14 +1541,18 @@ func (ec *EngineController) startRebuilding(e *longhorn.Engine, replica, addr st
 		}
 
 		// start rebuild
+		sizeRequested, err := ec.ds.IsEngineImageCLIAPIVersionEqualToOrLargerThan(e.Spec.EngineImage, 6)
+		if err != nil {
```
Comment (c3y1huang) [inline controller/engine_controller.go:1545]: Should log error.

**Review by c3y1huang — CHANGES_REQUESTED** [review]

**Comment by innobead** [comment]
> After discussion, this is not quite right, because new changes to engine operations should only be added to the proxy interface instead of the old deprecated binary call anymore.

The root cause has been clarified with @derekbit and @mantissahz . @derekbit will update the context.

File: controller/engine_controller.go (hunk 1 above)
Comment (derekbit) [reply controller/engine_controller.go]: Resubmit the PR. Please help review, thank you. :)

**Review by innobead — APPROVED** [review]

**Comment by innobead** [comment]
@derekbit Please backport to 1.3.2. Thanks!

**Comment by derekbit** [comment]
## Pre Ready-For-Testing Checklist
* [ ] Where is the reproduce steps/test steps documented?
The reproduce steps/test steps are at: https://github.com/longhorn/longhorn/issues/4705#issue-1405685731 

* ~~[ ] Is there a workaround for the issue? If so, where is it documented?~~
The workaround is at:

* [ ] Does the PR include the explanation for the fix or the feature?
https://github.com/longhorn/longhorn/issues/4705#issue-1405685731

* [ ] Have the backend code been merged (Manager, Engine, Instance Manager, BackupStore etc) (including `backport-needed/*`)?
The PR is at https://github.com/longhorn/longhorn-manager/pull/1523

* [ ] Which areas/issues this PR might have potential impacts on?
Area: replica add
Issues

**Pull request merged**